### PR TITLE
[SPARK-41508][CORE][SQL] Rename `_LEGACY_ERROR_TEMP_1180` to `UNEXPECTED_INPUT_TYPE` and remove `_LEGACY_ERROR_TEMP_1179`

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
The main changes of this pr as follows:
- Introduce `UNEXPECTED_INPUT_TYPE` instead of `_LEGACY_ERROR_TEMP_1180` and rename the method name from `incompatibleRangeInputDataTypeError` to `unexpectedInputDataTypeError`
- Refactor `Range#toLong` and `Range#toInt` to pass `paramIndex` to `UNEXPECTED_INPUT_TYPE` 
- Throw existing `AnalysisException` in `TableFunctionBuilder` instead of throwing a new `AnalysisException`
- Remove unused `_LEGACY_ERROR_TEMP_1179` and `_LEGACY_ERROR_TEMP_1180`

### Why are the changes needed?
Proper names of error classes to improve user experience with Spark SQL.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GitHub Actions

### CODE REVIEW

**Comment by LuciferYang** [comment]
Test first and will update pr description later

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -1152,6 +1152,11 @@
       "Found an unclosed bracketed comment. Please, append */ at the end of the comment."
     ]
   },
+  "UNEXPECTED_INPUT_TYPE" : {
+    "message" : [
+      "Parameter <paramIndex> requires the <requiredType> type, however <inputSql> has the type <inputType>."
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:1157]: This one is some kind of incomplete. `<paramIndex>` of what?

**Review by MaxGekk — APPROVED** [review]
Waiting for CI.

**Comment by LuciferYang** [comment]
GA passed

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @LuciferYang.

**Pull request closed without merging**